src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp

```diff
@@ -71,8 +72,26 @@ class ddscxx_serdata : public ddsi_serdata {
   bool& key_md5_hashed() { return m_key_md5_hashed; }
   const bool& key_md5_hashed() const { return m_key_md5_hashed; }
   void populate_hash();
-  T& getT() { return m_t; }
-  const T& getT() const { return m_t; }
+
+  T* getT()
+  {
+    if (!m_t)
+    {
+      m_t = new T();
+      switch (kind)
+      {
+      case SDK_KEY:
+        m_t->key_read(calc_offset(data(), 4), 0);
```

Comment:
Shouldn't there be a check for if the data field has been initialized?
This is the case when there is data being read, but also when it is being written?

Reply:
I don't understand this remark in this particular context. The only change made here is that getT() now returns a pointer to the sample, instead of a reference to the sample. Hence the '.' should be replaced by a '->' in the line you now quote. It could be that you mean this a a more general remark, but then again: how do you know if a sample is initialized since even a sample filled with 0's (the default) still counts as a valid sample.